### ISSUE TITLE
In mobile input composition, autocomplete won't update

### ISSUE THREAD

**Issue by jackycute** [codemirror/CodeMirror#3317]
![output_vybeg4](https://cloud.githubusercontent.com/assets/2639151/8038380/c5a7aeaa-0e35-11e5-8987-9338ff5938b9.gif)
Like above gif shown, when I input `ㄘㄜˋㄕˋ`, the autocomplete is correct.
But, if I try to delete some of the composition words, the autocomplete won't update.

The normal situation should like below:
![output_2c4dxw](https://cloud.githubusercontent.com/assets/2639151/8038379/c5a3c84e-0e35-11e5-9b9b-d6411b15aa62.gif)

Thanks for your help

**Comment by marijnh** [comment]
We are currently only applying the input to the actual document when the IME composition finishes. That means that addons, such as the completion addon, won't see updates as you are composing.

I might at some point change this, but for now, this is how it works.

**Comment by adrianheine** [comment]
I don't think we are going to change this, since we try to not disrupt the browser when doing composition. Right?

**Comment by marijnh** [comment]
The plan is to update the editor state on each input change. There _will_ be some trickiness about what you can do in these updates without causing the composition to be aborted (placing an inline widget in the composed code, for example, wouldn't work, but overlaying it over the editor should be fine).

I've opened https://github.com/codemirror/codemirror.next/issues/30 for this

**Comment by anashope** [comment]
J
